Replace getExperiment* and testExperiment* functions with getCrossAssociation

theotimeee opened the pull request:
Create new function `getCrossAssociation` to replace the following :

- getExperimentCrossAssociation
- getExperimentCrossCorrelation
- testExperimentCrossAssociation
- testExperimentCrossCorrelation

New function `getCrossAssociation` makes the significance test as default `(test_significance` default value is `TRUE`)

Commit:
rename getExperimentCrossAssociation to getCrossAssociation
deprecate getExperimentCrossAssociation
deprecate getExperimentCrossCorrelation
deprecate testExperimentCrossAssociation
deprecate testExperimentCrossCorrelation

Commit:
replace deprecated functions with new getCrossAssociation

Commit:
merge devel

Merge branch 'devel' of https://github.com/microbiome/mia into getCrossAssociation

# Conflicts:
#	man/deprecate.Rd

TuomasBorman commented on R/deprecate.R:
```diff
@@ -56,3 +56,83 @@ setMethod("taxonomyTree", signature = c(x = "SummarizedExperiment"),
+                .Deprecated(msg = paste0("'getExperimentCrossAssociation' is ",
+                                        "deprecated. Use ", 
+                                        "'getCrossAssociation' instead."))
+                getExperimentCrossAssociation(x, ...)
```
These functions are calling themselves

TuomasBorman replied:
Not the one that is new (getCrossAssociation)

TuomasBorman commented:
Looks good, the idea of this PR was to remove additional functions that we have only one function for cross-correlation. Moreover, the name of function was shortened

TuomasBorman commented:
test_significance could still be FALSE by default. The function uses base (or stats) R by default for calculating cross-correlation. The function that calculates correlation without significance testing is much faster

Commit:
correct test file

Commit:
set test_significance default value back to FALSE

Commit:
fix test file for getCrossAssociation

Commit:
merge devel

Merge branch 'devel' of https://github.com/microbiome/mia into getCrossAssociation

# Conflicts:
#	R/deprecate.R
#	man/deprecate.Rd

TuomasBorman commented on R/getCrossAssociation.R:
```diff
@@ -277,12 +274,12 @@ setMethod("getExperimentCrossAssociation", signature = c(x = "MultiAssayExperime
-#' @rdname getExperimentCrossAssociation
-#' @aliases getExperimentCrossCorrelation
+#' @rdname getCrossAssociation
+#' @aliases getCrossCorrelation
```
This line should be removed. There is no function called getCrossCorrelation

TuomasBorman commented on pkgdown/_pkgdown.yml:
```diff
@@ -46,7 +46,7 @@ reference:
   - getPrevalence
   - getPrevalentAbundance
   - getTopTaxa
-  - getExperimentCrossCorrelation
```
getCrossAssociation

Commit:
fix aliases and pkgdown file

Commit:
add export tag loadFromBiom

Commit:
Merge devel

Commit:
update NEWS and DESCRIPTION files

antagomir commented:
Some conflicts..?

antagomir commented:
Otherwise seems fine

Commit:
resolve conflicts and merge devel

Merge branch 'devel' of https://github.com/microbiome/mia into getCrossAssociation

# Conflicts:
#	DESCRIPTION
#	NEWS
#	R/deprecate.R
#	man/deprecate.Rd
#	pkgdown/_pkgdown.yml

Commit:
fix documentation deprecate file
